staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc_test.go

```diff
@@ -1585,6 +1618,45 @@ func TestToken(t *testing.T) {
 			}`, valid.Unix()),
 			wantErr: `oidc: verify token: oidc: expected audience in ["my-client" "random-client"] got ["not-my-client"]`,
 		},
+		{
+			name: "nuanced audience validation using claim validation rules",
```

Comment:
Can you also add this same test but with a token that fails?

Reply:
Done.